src/agent/src/rpc.rs

```diff
@@ -1906,6 +1917,91 @@ mod tests {
         assert!(result.is_err(), "expected add arp neighbors to fail");
     }
 
+    #[tokio::test]
+    async fn test_get_memory_info() {
+        #[derive(Debug)]
+        struct TestData<'a> {
+            // if None is provided, no files will be generated, else the data in the Option will populate the file
+            block_size_data: Option<&'a str>,
+            hotplug_probe_data: Option<&'a str>,
+
+            result: (u64, bool),
+            error_contains: &'a str,
```

Comment:
You could combine these into a `Result` as that would match the type returned by the function being tested.

```suggestion
            result: Result<(u64, bool)>,
```

That would allow you to create tests with values like:

```rust
            result: Ok((1234, true)),
            result: Ok((0, false)),
            result: Err(anyhow!("foo bar")),
```

For a few other examples simiilar to this, see:

- https://github.com/kata-containers/kata-containers/blob/main/src/agent/src/config.rs#L1228
- https://github.com/kata-containers/kata-containers/blob/main/src/agent/src/config.rs#L970

Reply:
One question with this, I like the `assert_result` macro, but it is currently defined in `config.rs`. I have copied it to `rpc.rs`, but is there a place for functions/macros shared between tests in different files? I looked around a bit but didn't see any.